components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -1145,9 +1145,10 @@ fn cast_string_as_time(
 ) -> Result<Option<Time>> {
     if let Some(val) = val {
         // Convert `val` to a string first and then parse it as a float string.
+        let val = String::from_utf8_lossy(val);
```

Comment:
I am a little confused here, where will the expected warning will produced.

Reply:
If it is not valid UTF-8 bytes, the converted string will contain a replacement character `U+FFFD`. This character cannot be parsed so the parser will generate the warning.

Reply:
I got it!